pkgs/by-name/co/comfyui/package.nix

```diff
@@ -0,0 +1,103 @@
+{ lib
+, python311
+, linkFarm
+, writers
+, writeTextFile
+, fetchFromGitHub
+, stdenv
+, torch ? python311.pkgs.torch
+, modelsPath ? "/var/lib/comfyui/models"
+, inputPath ? "/var/lib/comfyui/input"
+, outputPath ? "/var/lib/comfyui/output"
+, customNodes ? []
+}:
+
+let
+  modelPathsFile = writeTextFile {
+    name = "model_paths.yaml";
+    text = (lib.generators.toYAML {} {
+      comfyui = {
+        base_path = modelsPath;
+        checkpoints = "${modelsPath}/checkpoints";
+        clip = "${modelsPath}/clip";
+        clip_vision = "${modelsPath}/clip_vision";
+        configs = "${modelsPath}/configs";
+        controlnet = "${modelsPath}/controlnet";
+        embeddings = "${modelsPath}/embeddings";
+        loras = "${modelsPath}/loras";
+        upscale_models= "${modelsPath}/upscale_models";
+        vae = "${modelsPath}/vae";
+        vae_approx = "${modelsPath}/vae_approx";
+      };
+    });
+  };
+
+  pythonEnv = (python311.withPackages (ps: [ torch ] ++ [
+    ps.torchsde
+    ps.torchvision
+    ps.torchaudio
+    ps.transformers
+    ps.safetensors
+    ps.accelerate
+    ps.aiohttp
+    ps.einops
+    ps.pyyaml
+    ps.pillow
+    ps.scipy
+    ps.psutil
+    ps.tqdm
+  ] ++ (builtins.concatMap (node: node.dependencies) customNodes)));
+
+  executable = writers.writeDashBin "comfyui" ''
+    cd $out && ${pythonEnv}/bin/python comfyui "$@"
+  '';
+
+  customNodesCollection = (
+    linkFarm "comfyui-custom-nodes" (builtins.map (pkg: { name = pkg.pname; path = pkg; }) customNodes)
+  );
+in stdenv.mkDerivation rec {
+  pname = "comfyui";
+  version = "unstable-2023-11-19";
+
+  src = fetchFromGitHub {
+    owner = "comfyanonymous";
+    repo = "ComfyUI";
+    rev = "d9d8702d8dd2337c64610633f5df2dcd402379a8";
+    hash = "sha256-icfy14uNqVoaxR8Rqw1d5+k0p3WQYzcGegrgNxt/nOc=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+    echo "Preparing bin folder"
+    mkdir -p $out/bin/
+    echo "Copying comfyui files"
+    cp -r $src/comfy $out/
+    cp -r $src/comfy_extras $out/
```

Comment:
If you like, here's a potential alternative to the custom `installPhase`:

- https://github.com/SomeoneSerge/pkgs/blob/f28af3c0d60500697a987794b2b1e64f70cfd3b7/python-packages/by-name/co/comfy-ui/package.nix#L55
- https://github.com/SomeoneSerge/pkgs/blob/f28af3c0d60500697a987794b2b1e64f70cfd3b7/python-packages/by-name/co/comfy-ui/pyproject.toml

Reply:
I like this alternative much better.
Even better, submit a patch along with what SomeoneSerge said in the other comment and just fetch that patch so we don't have to include extra files in nixpkgs.